dkg/pedersen/mod.go

```diff
@@ -270,6 +330,155 @@ func (a *Actor) Decrypt(K, C kyber.Point) ([]byte, error) {
 	return decryptedMessage, nil
 }
 
+// VerifiableDecrypt implements dkg.Actor. It does as Decrypt() but in addition
+// it checks whether the decryption proofs are valid.
+//
+// See https://arxiv.org/pdf/2205.08529.pdf / section 5.4 Protocol / step 3
+func (a *Actor) VerifiableDecrypt(ciphertexts []types.Ciphertext) ([][]byte, error) {
```

Comment:
After this PR: this could also benefit from being refactored into smaller functions